src/Migration.php

```diff
@@ -382,9 +382,9 @@ private function fieldFormat($type, $default_value, $nodefault = false): string
      * @param string $table   Table name
      * @param string $field   Field name
      * @param string $type    Field type, @see Migration::fieldFormat()
-     * @param array  $options Options:
-     *                         - update    : if not empty = value of $field (must be protected)
-     *                         - condition : if needed
+     * @param array{update?: string, condition?: string, value?: string, nodefault?: bool, comment?: string, first?: string, after?: string, null?: bool} $options
```

Comment:
Will phpstan (in higher levels) be happy with that notation?

Reply:
Yes. This is the best way to document options. Even with the current level, it can detect misusage of options (e.g. an undocumented option, a wrong type used in an option, ...).